transaction/core/src/tx_error.rs

```diff
@@ -69,10 +69,14 @@ pub enum NewMemoError {
     InvalidRecipient,
     /// Multiple outputs are not supported
     MultipleOutputs,
+    /// Missing required input to build the memo
+    MissingInput(String),
     /// Missing output
     MissingOutput,
     /// Mixed Token Ids are not supported in these memos
     MixedTokenIds,
+    /// Gift Code Destination Memos are not supported
+    GiftCodeDesintationMemoNotAllowed,
```

Comment:
This we'll suss out soon, the design is up in there about whether a destination memo for RTH should be written by the builder or whether it should be handed off to the RTH builder. Once this is sussed out I'll more properly name it :)

Reply:
Probably just DestinationMemoNotAllowed - I feel like since there are now a variety of memo types (most frequently used Change/Destination) including it in. "NotAllowed" could allude to a lot of things, and I think there's no harm in being extra clear.